### PR TITLE
add keyboard-discipline

### PR DESCRIPTION
**Pull request by coseyfannitutti**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by noroadsleft** [comment]
Any particular reason you haven't put this in your `keyboards/coseyfannitutti/` folder?

**Comment by coseyfannitutti** [comment]
> 
> 
> Any particular reason you haven't put this in your `keyboards/coseyfannitutti/` folder?

No, I should have done that. That can be changed, and the example flash command in the readme can be changed to reflect that.
Do I need to change that on my end and submit another pull request for that to happen?

**Comment by noroadsleft** [comment]
You don't need to open a new pull request. Just check out your master branch (`git checkout master`), then run:

```shell
git mv keyboards/discipline/ keyboards/coseyfannitutti/discipline/
git commit -m "[commit message]"
git push
```

and that will update this pull request.

File: keyboards/coseyfannitutti/discipline/discipline.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  K40, K41, K42,                K46,           K49, K4A, K4B, K4C, K4D, K4E \
+) { \
+{ K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E }, \
+{ K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E }, \
+{ K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, _x_, K2D, K2E }, \
+{ K30, _x_, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, K3E }, \
+{ K40, K41, K42, _x_, _x_, _x_, K46, _x_, _x_, K49, K4A, K4B, K4C, K4D, K4E} \
+}
+
+#define LAYOUT_67_ansi( \
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/discipline.h]: Just trying to keep the macro names consistent for this board.

```suggestion
#define LAYOUT_65_ansi_2_right_mods( \
```

File: keyboards/coseyfannitutti/discipline/info.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "keyboard_name": "DISCIPLINE", 
+    "url": "https://github.com/coseyfannitutti/discipline", 
+    "maintainer": "coseyfannitutti", 
+    "width": 16, 
+    "height": 5, 
+    "layouts": {
+        "LAYOUT_68_ansi": {
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/info.json]:
```suggestion
        "LAYOUT_65_ansi": {
```

File: keyboards/coseyfannitutti/discipline/info.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        },
+        "LAYOUT_67_ansi": {
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/info.json]:
```suggestion
        "LAYOUT_65_ansi_2_right_mods": {
```

File: keyboards/coseyfannitutti/discipline/keymaps/67_ansi/keymap.c [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+   * | Caps    |     |     |     |     |     |     |     |     |     |     | INS |    Enter  | HOME|
+   * |---------------------------------------------------------------------------------------------+
+   * | Shift     |     |     |     |     |     |     |     |     |     |     |  Shift  |VOLUP| END |
+   * |---------------------------------------------------------------------------------------------+
+   * | Ctrl |  Win  |  Alt  |                                   | RAlt| FN |||||||  L  |VOLDN|  R  |
+   * '---------------------------------------------------------------------------------------------'
+   */
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BL] = LAYOUT_67_ansi(
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/keymaps/67_ansi/keymap.c]:
```suggestion
  [_BL] = LAYOUT_65_ansi_2_right_mods(
```

File: keyboards/coseyfannitutti/discipline/keymaps/67_ansi/keymap.c [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_GRV,
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_DEL,
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,            KC_PGUP,
+      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    KC_PGDN,
+      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RALT, MO(_FL),           KC_LEFT, KC_DOWN,  KC_RIGHT),
+
+  [_FL] = LAYOUT_67_ansi(
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/keymaps/67_ansi/keymap.c]:
```suggestion
  [_FL] = LAYOUT_65_ansi_2_right_mods(
```

File: keyboards/coseyfannitutti/discipline/keymaps/coseyfannitutti/keymap.c [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+   * | Caps    |     |     |     |     |     |     |     |     |     |     | INS |    Enter  | HOME|
+   * |---------------------------------------------------------------------------------------------+
+   * | Shift     |     |     |     |     |     |     |     |     |     |     |  Shift  |VOLUP| END |
+   * |---------------------------------------------------------------------------------------------+
+   * | Ctrl |  Win  |  Alt  |                                   | RAlt| FN |||||||  L  |VOLDN|  R  |
+   * '---------------------------------------------------------------------------------------------'
+   */
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BL] = LAYOUT_67_ansi(
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/keymaps/coseyfannitutti/keymap.c]:
```suggestion
  [_BL] = LAYOUT_65_ansi_2_right_mods(
```

File: keyboards/coseyfannitutti/discipline/keymaps/coseyfannitutti/keymap.c [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_DEL,
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_PGUP,
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,            KC_PGDN,
+      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    MO(_FL),
+      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RALT, MO(_FL),           KC_LEFT, KC_DOWN,  KC_RIGHT),
+
+  [_FL] = LAYOUT_67_ansi(
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/keymaps/coseyfannitutti/keymap.c]:
```suggestion
  [_FL] = LAYOUT_65_ansi_2_right_mods(
```

File: keyboards/coseyfannitutti/discipline/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+   * | Caps    |     |     |     |     |     |     |     |     |     |     | INS |    Enter  | HOME|
+   * |---------------------------------------------------------------------------------------------+
+   * | Shift     |     |     |     |     |     |     |     |     |     |     |  Shift  |VOLUP| END |
+   * |---------------------------------------------------------------------------------------------+
+   * | Ctrl |  Win  |  Alt  |                                   | RAlt| FN |RCtrl|  L  |VOLDN|  R  |
+   * '---------------------------------------------------------------------------------------------'
+   */
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BL] = LAYOUT_68_ansi(
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/keymaps/default/keymap.c]:
```suggestion
  [_BL] = LAYOUT_65_ansi(
```

File: keyboards/coseyfannitutti/discipline/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+      KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_GRV,
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_DEL,
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,            KC_PGUP,
+      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    KC_PGDN,
+      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RALT, MO(_FL), KC_RCTL,  KC_LEFT, KC_DOWN,  KC_RIGHT),
+
+  [_FL] = LAYOUT_68_ansi(
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/keymaps/default/keymap.c]:
```suggestion
  [_FL] = LAYOUT_65_ansi(
```

File: keyboards/coseyfannitutti/discipline/readme.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+A 65% keyboard that can be assembled with only through hole components, including usb type-c
+
+Keyboard Maintainer: [coseyfannitutti](https://github.com/coseyfannitutti)  
+Hardware Supported: DISCIPLINE, atmega32a  
+Hardware Availability: [cftkb.com](http://www.cftkb.com), [GitHub](https://github.com/coseyfannitutti/discipline)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make coseyfannitutti/discipline:program
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/readme.md]: This is actually `<keyboard>:<keymap>:<program>`, with `<program>` being optional.

```suggestion
    make coseyfannitutti/discipline:default:program
```

File: keyboards/coseyfannitutti/discipline/rules.mk [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+
+# unsupported features for now
+NO_UART = yes
+NO_SUSPEND_POWER_DOWN = yes
```
Comment (noroadsleft) [inline keyboards/coseyfannitutti/discipline/rules.mk]: And this change enables support for the Community Layouts feature.

```suggestion
NO_SUSPEND_POWER_DOWN = yes

LAYOUTS = 65_ansi
```

File: keyboards/coseyfannitutti/discipline/discipline.c [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "discipline.h"
+
+uint8_t send_data;
```
Comment (drashna) [inline keyboards/coseyfannitutti/discipline/discipline.c]: I don't think that this is actually used anywhere.  If that's the case, it should probably be removed.

```suggestion
```

**Pull request merged**